build.cake

```
@@ -0,0 +1,268 @@
+
+// Custom settings for VSTest
+var VSTestCustomSettings = new VSTestSettings()
+{
+	ArgumentCustomization = args => args.Append("/TestAdapterPath:" + ADAPTER_BIN_DIR)
+};
+
+//////////////////////////////////////////////////////////////////////
+// CLEAN
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Does(() =>
+{
+    CleanDirectory(ADAPTER_BIN_DIR);
+    CleanDirectory(TEST_BIN_DIR);
+    CleanDirectory(INSTALL_BIN_DIR);
+	CleanDirectory(DEMO_BIN_DIR);
+});
+
+
+//////////////////////////////////////////////////////////////////////
+// INITIALIZE FOR BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("InitializeBuild")
+    .Does(() =>
+{
+    NuGetRestore(ADAPTER_SOLUTION);
+	NuGetRestore(DEMO_SOLUTION);
+
+	if (BuildSystem.IsRunningOnAppVeyor)
+	{
+		var tag = AppVeyor.Environment.Repository.Tag;
+
+		if (tag.IsTag)
+		{
+			packageVersion = tag.Name;
+		}
+		else
+		{
+			var buildNumber = AppVeyor.Environment.Build.Number;
+			packageVersion = version + "-CI-" + buildNumber + dbgSuffix;
+			if (AppVeyor.Environment.PullRequest.IsPullRequest)
+				packageVersion += "-PR-" + AppVeyor.Environment.PullRequest.Number;
+			else
+				packageVersion += "-" + AppVeyor.Environment.Repository.Branch;
+		}
+
+		AppVeyor.UpdateBuildVersion(packageVersion);
+	}
+});
+
+//////////////////////////////////////////////////////////////////////
+// BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("Build")
+    .IsDependentOn("InitializeBuild")
+    .Does(() =>
+    {
+		BuildSolution(ADAPTER_SOLUTION, configuration);
+		BuildSolution(DEMO_SOLUTION, configuration);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEST
+//////////////////////////////////////////////////////////////////////
+
+Task("TestAdapterUsingConsole")
+	.IsDependentOn("Build")
+	.Does(() =>
+	{
```

I may be wrong, but from memory if we run the console as a process, we need to throw an exception based on the return code. I don't think this will break the build on a failed test as it's currently written - although not setup to pull it and try right now I'm afraid! 

Would perhaps be nice to log all the failed tests in a list like the main nunit/nunit script, given there's multiple tests suites being run. I think this will currently stop on the first failed suite?